main/src/ca/uwaterloo/flix/language/errors/StratificationError.scala

```diff
@@ -17,14 +17,30 @@
 package ca.uwaterloo.flix.language.errors
 
 import ca.uwaterloo.flix.language.CompilationError
+import ca.uwaterloo.flix.language.ast.TypedAst.Predicate.Body.{Filter, Loop, Negative, Positive}
+import ca.uwaterloo.flix.language.ast.TypedAst.Stratum
 import ca.uwaterloo.flix.language.ast._
+import ca.uwaterloo.flix.util.vt.VirtualString._
 import ca.uwaterloo.flix.util.vt.VirtualTerminal
 
 /**
   * An error raised to indicate that a property is violated.
```

Comment:
Here's a sample error message:
```
-- Stratification Error -------------------------------------------------- TestStratifiedNegation.flix

>> Stratification Error in constraints
It was triggered by the rule: Foo(...) :- !Foo(...)
At TestStratifiedNegation.flix:19:11
```

I suppose one of the benefits of doing Negative Cycle checking would be that if you detect a cycle, you could reconstruct what that cycle was and pass it as an error message, which would be more helpful than listing only the offending rule. Perhaps it would be better to partially bring back the negative cycle check and run it if we find a stratification error?